azuredevops/internal/service/git/resource_git_repository.go

```
@@ -282,18 +290,68 @@ func resourceGitRepositoryUpdate(d *schema.ResourceData, m interface{}) error {
+	repoActual, err := gitRepositoryRead(clients, repo.Id.String(), *repo.Name, projectID.String())
+	if err != nil {
+		if utils.ResponseWasNotFound(err) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error looking up repository with ID %s and Name %s. Error: %v", *repo.Id, *repo.Name, err)
+	}
+
+	if *repoActual.IsDisabled {
+		err = updateIsDisabledGitRepository(clients, repo.Id.String(), projectID.String(), false)
+		if err != nil {
+			return fmt.Errorf(" enabling created repository : %+v", err)
+		}
+	}
+
 	_, err = updateGitRepository(clients, repo, projectID)
 	if err != nil {
 		return fmt.Errorf("Error updating repository in Azure DevOps: %+v", err)
 	}
 
+	// Disabling the repo after an update if it should be disabled
+	if v := d.Get("disabled").(bool); v {
+		err = updateIsDisabledGitRepository(clients, repo.Id.String(), projectID.String(), true)
+		if err != nil {
+			return fmt.Errorf(" disabling created repository : %+v", err)
+		}
+	}
+
 	return resourceGitRepositoryRead(d, m)
 }
 
 func resourceGitRepositoryDelete(d *schema.ResourceData, m interface{}) error {
 	repoID := d.Id()
+	repoName := d.Get("name").(string)
+	projectID := d.Get("project_id").(string)
 	clients := m.(*client.AggregatedClient)
-	err := deleteGitRepository(clients, repoID)
+
+	uuid, err := uuid.Parse(repoID)
+	if err != nil {
+		return fmt.Errorf("Invalid repositoryId UUID: %s", repoID)
+	}
+
+	// you cannot delete a disabled repo. We must enable in order to delete
+	repoActual, err := gitRepositoryRead(clients, repoID, repoName, projectID)
+	if err != nil {
+		if utils.ResponseWasNotFound(err) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error looking up repository with ID %s and Name %s. Error: %v", repoID, repoName, err)
+	}
+
+	if *repoActual.IsDisabled {
+		err = updateIsDisabledGitRepository(clients, repoID, projectID, false)
+		if err != nil {
+			return fmt.Errorf(" enabling created repository : %+v", err)
+		}
+	}
```

It's better to raise an error than to enable the repo and update it.

---

@xuzhang3 I have updated the code per your recommendation to error instead of simply enabling the repo. The code is ready for another review.